lib/src/front_end/piece_factory.dart

```diff
@@ -189,83 +186,80 @@ mixin PieceFactory implements CommentWriter {
         visit(node.thenStatement);
       }
 
-      var thenStatement = writer.pop();
-      writer.split();
+      var thenStatement = pieces.split();
       piece.add(condition, thenStatement, isBlock: node.thenStatement is Block);
 
       switch (node.elseStatement) {
         case IfStatement elseIf:
           // Hit an else-if, so flatten it into the chain with the `else`
           // becoming part of the next section's header.
           token(node.elseKeyword);
-          writer.space();
+          space();
           traverse(elseIf);
 
         case var elseStatement?:
           // Any other kind of else body ends the chain, with the header for
           // the last section just being the `else` keyword.
           token(node.elseKeyword);
-          var header = writer.pop();
-          writer.split();
+          var header = pieces.split();
 
           visit(elseStatement);
-          var statement = writer.pop();
-          writer.split();
+          var statement = pieces.pop();
```

Comment:
Lost a call to `split()` here. Did any tests fail? Any tests we need to add?

```suggestion
          var statement = pieces.split();
```

Reply:
For the trailing chunk of code owned by some AST node, it often doesn't matter whether you capture it with `pop()` or `split()`. The distinction is only visible if some *surrounding* AST node wants to append more code directly that trailing node.

That's very common for expressions (think the `;` in an expression statement). But it's less common for statements.

In this case, both `pop()` and `split()` pass all the tests, and I can't offhand think of any place you could stuff an if statement where the distinction would matter. I picked `pop()` because that's the "friendliest" default way to grab the last chunk of code for a node: it lets the caller decide whether or not it wants to append more code to it.